src/openms/source/FORMAT/QcMLFile.cpp

```diff
@@ -1038,8 +1049,1023 @@ namespace OpenMS
     }
   }
 
-  void QcMLFile::store(const String& filename) const
+  double getMassDifference(double theo_mz, double exp_mz, bool use_ppm)
+  {
+    double error(exp_mz - theo_mz);
+    if (use_ppm)
+    {
+      error = error / (theo_mz * (double)1e-6);
+      //~ error = (1-exp_mz/theo_mz) * (double)1e6;
+    }
+    return error;
+  }
+
+  float calculateSNmedian(MSSpectrum spec, bool norm = true)
```

Comment:
I would use const MSSpectrum& spec here and copy the intensities into a vector<double> and use our median method in math or statistics namespace

Reply:
actually I wonder if we don't have SNmedian estimation in OpenMS? can you check. I think we might have several implementations

Reply:
I could not find an already existing SNmedian function, however, I used the Math::median() function here